cmd/podman/save.go

```diff
@@ -118,14 +119,26 @@ func saveCmd(c *cli.Context) error {
 			return err
 		}
 	}
-	newImage, err := runtime.ImageRuntime().NewFromLocal(args[0])
+	source := args[0]
+	newImage, err := runtime.ImageRuntime().NewFromLocal(source)
 	if err != nil {
 		return err
 	}
 	dest := dst
 	// need dest to be in the format transport:path:reference for the following transports
-	if (strings.Contains(dst, libpod.OCIArchive) || strings.Contains(dst, libpod.DockerArchive)) && !strings.Contains(newImage.ID(), args[0]) {
-		dest = dst + ":" + args[0]
+	if (strings.Contains(dst, libpod.OCIArchive) || strings.Contains(dst, libpod.DockerArchive)) && !strings.Contains(newImage.ID(), source) {
+		prepend := ""
+		if !strings.Contains(source, libpodImage.DefaultLocalRepo) {
```

Comment:
> The only way I can think of to differentiate a registry and a repository is the presence of the string ("example.com" in this case) in registries.conf.

`registries.conf` contains almost none of the registries that exist in the world.  Ideally, the difference should be ~syntactic, i.e. it should be possible to decide based just on the string contents instead of trying to check whether a registry / repository happens to exist with that name; I have linked to such implementations in #1140.